h2integrate/core/h2integrate_model.py

```diff
@@ -65,19 +63,134 @@ def __init__(self, config_file):
+            - If `config_file` is a dict: Uses it directly as the main configuration
+            - If `config_file` is a path: Uses `get_path()` to resolve and load the YAML file
+              from multiple search locations (absolute path, relative to CWD, relative to
+              H2Integrate package)
+
+            For each component config (driver_config, technology_config, plant_config):
+            - If the config value is a dict: Validates it directly using the appropriate
+              validation function (`load_driver_yaml`, `load_tech_yaml`, `load_plant_yaml`)
+            - If the config value is a path string:
+                - When main config was loaded from file: Uses `find_file()` to search
+                  relative to the main config file's directory first, then falls back
+                  to other search locations (CWD, H2Integrate package, glob patterns)
+                - When main config was provided as dict: Uses `get_path()` with standard
+                  search strategy (absolute, relative to CWD, relative to H2Integrate package)
+
+        Sets:
+            self.name (str): Name of the system from main config
+            self.system_summary (str): Summary description from main config
+            self.driver_config (dict): Validated driver configuration
+            self.technology_config (dict): Validated technology configuration
+            self.plant_config (dict): Validated plant configuration
+            self.driver_config_path (Path | None): Path to driver config file (None if dict)
+            self.tech_config_path (Path | None): Path to technology config file (None if dict)
+            self.plant_config_path (Path | None): Path to plant config file (None if dict)
+            self.tech_parent_path (Path | None): Parent directory of technology config file
+            self.plant_parent_path (Path | None): Parent directory of plant config file
+
+        Note:
+            The parent path attributes (tech_parent_path, plant_parent_path) are used later
+            for resolving relative paths to custom models and other referenced files within
+            the technology and plant configurations.
+
+        Example:
+            >>> # Using filepaths
+            >>> model = H2IntegrateModel("main_config.yaml")
+
+            >>> # Using mixed dict and filepaths
+            >>> config = {
+            ...     "name": "my_system",
+            ...     "driver_config": "driver.yaml",
+            ...     "technology_config": {"technologies": {...}},
+            ...     "plant_config": "plant.yaml",
+            ... }
+            >>> model = H2IntegrateModel(config)
+        """
+        config_path = None
+        tech_path = None
+        driver_path = None
+        plant_path = None
+        tech_parent_path = None
+        plant_parent_path = None
+
+        if isinstance(config_file, dict):
+            # Use dictionary directly without filepath context
+            config = config_file
```

Comment:
Non-blocking nitpick for clarity: maybe it makes sense to rename `config_file` to something like `config_input`, which can then become the `config` dict directly. Because the name `config_file` has `file` in it, we could be more clear that it might be a dict by changing its name.